rust/yarp-sys/build.rs

```diff
@@ -0,0 +1,168 @@
+        // .allowlist_type("yp_string_t")
+        .blocklist_type(r#"^yp_string_t\S*"#)
+        .allowlist_type("yp_string_list_t")
+        .allowlist_type("yp_token_type_t")
+        // Enums
+        .rustified_non_exhaustive_enum("yp_comment_type_t")
+        .rustified_non_exhaustive_enum("yp_context_t")
+        .rustified_non_exhaustive_enum("yp_heredoc_indent_t")
+        .rustified_non_exhaustive_enum("yp_heredoc_quote_t")
+        .rustified_non_exhaustive_enum("yp_lex_mode_t")
+        .rustified_non_exhaustive_enum("yp_lex_state_t")
+        .rustified_non_exhaustive_enum("yp_node_type")
+        .rustified_non_exhaustive_enum("yp_pack_encoding")
+        .rustified_non_exhaustive_enum("yp_pack_endian")
+        .rustified_non_exhaustive_enum("yp_pack_length_type")
+        .rustified_non_exhaustive_enum("yp_pack_result")
+        .rustified_non_exhaustive_enum("yp_pack_signed")
+        .rustified_non_exhaustive_enum("yp_pack_size")
+        .rustified_non_exhaustive_enum("yp_pack_type")
+        .rustified_non_exhaustive_enum("yp_pack_variant")
+        .rustified_non_exhaustive_enum("yp_token_type")
+        .rustified_non_exhaustive_enum("yp_unescape_type_t")
+        // Functions
+        .allowlist_function("yp_buffer_init")
+        .allowlist_function("yp_buffer_free")
+        .allowlist_function("yp_node_destroy")
+        .allowlist_function("yp_list_empty_p")
+        .allowlist_function("yp_list_free")
+        .allowlist_function("yp_list_init")
+        .allowlist_function("yp_node_memsize")
+        .allowlist_function("yp_pack_parse")
+        .allowlist_function("yp_parse")
+        .allowlist_function("yp_parse_serialize")
+        .allowlist_function("yp_parser_free")
+        .allowlist_function("yp_parser_init")
+        .allowlist_function("yp_parser_register_encoding_changed_callback")
+        .allowlist_function("yp_parser_register_encoding_decode_callback")
+        .allowlist_function("yp_prettyprint")
+        .allowlist_function("yp_regexp_named_capture_group_names")
+        .allowlist_function("yp_serialize")
+        .allowlist_function("yp_size_to_native")
+        .allowlist_function("yp_string_free")
+        .allowlist_function("yp_string_length")
+        .allowlist_function("yp_string_source")
+        .allowlist_function("yp_string_list_init")
+        .allowlist_function("yp_string_list_free")
+        .allowlist_function("yp_token_type_to_str")
+        .allowlist_function("yp_unescape_calculate_difference")
+        .allowlist_function("yp_unescape_manipulate_string")
+        .allowlist_function("yp_version")
+        // Vars
+        .allowlist_var(r#"^yp_encoding\S+"#)
+        .generate()
+        .expect("Unable to generate yarp bindings")
+}
+
+/// Write the bindings to the $OUT_DIR/bindings.rs file. We'll pull these into the actual library
+/// in `src/lib.rs`.
+fn write_bindngs(bindings: bindgen::Bindings) {
+    let out_path = PathBuf::from(std::env::var("OUT_DIR").unwrap());
```

Comment:
```suggestion
    let out_path = PathBuf::from(std::env::var_os("OUT_DIR").unwrap());
```

Reply:
Similar to my question above, if you want to ensure UTF-8, it'd seem you'd want to use `var()` here, not `var_os()`; why allow non-UTF8 encoding from an env var here, then fail later when you're outputting the `PathBuf` to `stdin`? I'd think if you really want to ensure UTF-8 all around (which I'm still unsure of why you would in these cases), you'd use `var()` and `Path::to_str()`. String encodings are not my expertise, though, so please explain if you have a min!

Reply:
this one actually is to _avoid_ needing to care about UTF-8 validation. We go from an env var to a pathbuf directly to a `std::fs` API. Using `var_os` here means the entire call chain never once does a UTF-8 validation.

In build scripts and when dealing with path and platform strings in general, my preference is to maintain the native representation for as long as possible, only converting to UTF-8 if there are no other options. Printing to stdout with `println!` and passing to bindgen APIs that take `Into<String>` are such cases where there is no choice but to convert to UTF-8, so we might as well assert that we, in fact, do.

Reply:
Ah yeah, that makes sense. Nice!